cpu/kinetis_common/uart.c

```diff
@@ -0,0 +1,360 @@
+            UART_0_DEV->C2 &= ~(1 << UART_C2_TIE_SHIFT);
+            break;
+#endif
+#if UART_1_EN
+
+        case UART_1:
+            UART_1_DEV->C2 &= ~(1 << UART_C2_TIE_SHIFT);
+            break;
+#endif
+
+        default:
+            break;
+    }
+}
+
+int uart_write(uart_t uart, char data)
+{
+    switch (uart) {
+#if UART_0_EN
+
+        case UART_0:
+            if (UART_0_DEV->S1 & UART_S1_TDRE_MASK) {
+                UART_0_DEV->D = (uint8_t)data;
+            }
+
+            break;
+#endif
+#if UART_1_EN
+
+        case UART_1:
+            if (UART_1_DEV->S1 & UART_S1_TDRE_MASK) {
+                UART_1_DEV->D = (uint8_t)data;
+            }
+
+            break;
+#endif
+
+        default:
+            return -2;
+            break;
+    }
+
+    return 0;
+}
+
+int uart_read_blocking(uart_t uart, char *data)
+{
+    switch (uart) {
+#if UART_0_EN
+
+        case UART_0:
+            while (!(UART_0_DEV->S1 & UART_S1_RDRF_MASK));
+
+            *data = (char)UART_0_DEV->D;
+            break;
+#endif
+#if UART_1_EN
+
+        case UART_1:
+            while (!(UART_1_DEV->S1 & UART_S1_RDRF_MASK));
+
+            *data = (char)UART_1_DEV->D;
+            break;
+#endif
+
+        default:
+            return -2;
+            break;
+    }
+
+    return 1;
+}
+
+int uart_write_blocking(uart_t uart, char data)
+{
+    switch (uart) {
+#if UART_0_EN
+
+        case UART_0:
+            while (!(UART_0_DEV->S1 & UART_S1_TDRE_MASK));
+
+            UART_0_DEV->D = (uint8_t)data;
+            break;
+#endif
+#if UART_1_EN
+
+        case UART_1:
+            while (!(UART_1_DEV->S1 & UART_S1_TDRE_MASK));
+
+            UART_1_DEV->D = (uint8_t)data;
+            break;
+#endif
+
+        default:
+            return -2;
+            break;
+    }
+
+    return 1;
+}
+
+#if UART_0_EN
+void UART_0_ISR(void)
+{
+    irq_handler(UART_0, UART_0_DEV);
+}
+#endif
+
+#if UART_1_EN
+void UART_1_ISR(void)
+{
+    irq_handler(UART_1, UART_1_DEV);
+}
+#endif
+
+static inline void irq_handler(uart_t uartnum, KINETIS_UART *dev)
+{
+    /*
+    * On Cortex-M0, it happens that S1 is read with LDR
+    * instruction instead of LDRB. This will read the data register
+    * at the same time and arrived byte will be lost. Maybe it's a GCC bug.
+    *
+    * Observed with: arm-none-eabi-gcc (4.8.3-8+..)
+    * It does not happen with: arm-none-eabi-gcc (4.8.3-9+11)
+    */
+
+    if (dev->S1 & UART_S1_RDRF_MASK) {
+        /* RDRF flag will be cleared when dev-D was read */
+        uint8_t data = dev->D;
+
+        if (config[uartnum].rx_cb != NULL) {
+            config[uartnum].rx_cb(config[uartnum].arg, data);
+        }
+    }
+
+    if (dev->S1 & UART_S1_TDRE_MASK) {
+        if (config[uartnum].tx_cb == NULL) {
+            dev->C2 &= ~(UART_C2_TIE_MASK);
+        }
+        else if (config[uartnum].tx_cb(config[uartnum].arg) == 0) {
+            dev->C2 &= ~(UART_C2_TIE_MASK);
+        }
+    }
+
+#ifndef KINETIS_UART_ADVANCED
```

Comment:
replace with `#if !(KINETIS_UART_ADVANCED)` or  `#if KINETIS_UART_ADVANCED == 0` (undefined will evaluate to zero), then we can explicitly set it to zero if necessary to avoid the default definition at the top of this file.